### ISSUE TITLE
`tuist lint code` reports violations in `Tuist/Dependencies`

### ISSUE THREAD

**Issue by nhaarman** [tuist/tuist#3964]
**Describe the bug**
When running `tuist lint code`, the command fails because there are violations in the `Tuist/Dependencies` folder.

Tuist version 2.6.0

**Comment by luispadron** [comment]
This is actually working as expected, albeit maybe not as intuitive since its linting the code for dependencies as well. 

When using `Dependencies.swift` those targets are also added to the project graph which means when using `tuist lint code` without any arguments, **all** of the targets will be linted.

You can do `tuist lint code <my-target>` to lint a specific target:

```
OVERVIEW: Lints the code of your projects using Swiftlint.

USAGE: tuist lint code [--path <path>] [<target>] [--strict]

ARGUMENTS:
  <target>                The target to be linted. When not specified all the
                          targets of the graph are linted.

OPTIONS:
  -p, --path <path>       The path to the directory that contains the workspace
                          or project whose code will be linted.
  --strict                Fails on warnings.
  -h, --help              Show help information.

```

@tuist/core Thoughts on perhaps filtering out dependencies from the lint command? Im not sure when it would be useful to lint dependency code.

**Comment by luispadron** [comment]
This is actually working as expected, albeit maybe not as intuitive since its linting the code for dependencies as well. 

When using `Dependencies.swift` those targets are also added to the project graph which means when using `tuist lint code` without any arguments, **all** of the targets will be linted.

You can do `tuist lint code <my-target>` to lint a specific target:

```
OVERVIEW: Lints the code of your projects using Swiftlint.

USAGE: tuist lint code [--path <path>] [<target>] [--strict]

ARGUMENTS:
  <target>                The target to be linted. When not specified all the
                          targets of the graph are linted.

OPTIONS:
  -p, --path <path>       The path to the directory that contains the workspace
                          or project whose code will be linted.
  --strict                Fails on warnings.
  -h, --help              Show help information.

```

@tuist/core Thoughts on perhaps filtering out dependencies from the lint command? Im not sure when it would be useful to lint dependency code.

**Comment by danyf90** [comment]
I agree by default they shouldn't be linked, as the focus by default imports them as binaries

**Comment by danyf90** [comment]
Cc @laxmorek who is implementing the new plugin for it

**Comment by laxmorek** [comment]
The `tuist lint code` command is going to be deprecated (removed?) in version 3.0. 

Im moving SwiftLint and linting logic to a separate plugin that will be accessible via the new plugins system.

If the issue is not a very big problem I would wait for our new approach.

PS. On the other hand filtering out dependencies from the lint command should be a hard task to do. 🤔 @nhaarman would you like to try to contribute? 

Here is a place that fires listing logic: [click](https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Services/Lint/LintCodeService.swift) and here is a line that will be invited when no target is provided: [click](https://github.com/tuist/tuist/blob/eb00251cc134333a02f62e8d140e9f7353f4cd46/Sources/TuistKit/Services/Lint/LintCodeService.swift#L90).
